third_party/move/move-compiler-v2/src/pipeline/reference_safety_processor.rs

```diff
@@ -1012,7 +1012,7 @@ impl<'env, 'state> LifetimeAnalysisStep<'env, 'state> {
     fn check_borrow_safety(&mut self, temps_vec: &[TempIndex]) {
         // First check direct duplicates
         for (i, temp) in temps_vec.iter().enumerate() {
-            if temps_vec[i + 1..].contains(temp) {
+            if self.ty(*temp).is_mutable_reference() && temps_vec[i + 1..].contains(temp) {
                 self.exclusive_access_direct_dup_error(*temp)
```

Comment:
If a test case corresponding to this (eg, call to a function with multiple dup args that are non-mut references) has not been added, please do.

Reply:
Done, new test is `duplicate_use.move`